packages/types/src/interfaces/payment/runtime.ts

```diff
@@ -39,25 +39,67 @@ const V1_V2_SHARED_CALL: Record<string, DefinitionCall> = {
   }
 };
 
+const V2_V3_SHARED_QUERY_INFO: Record<string, DefinitionCall> = {
+  query_call_info: {
+    description: 'The call info',
+    params: [
+      {
+        name: 'call',
+        type: 'Call'
+      },
+      {
+        name: 'len',
+        type: 'u32'
+      }
+    ],
+    type: 'RuntimeDispatchInfo'
+  }
+};
+
+const V3_QUERY_WEIGHT_TO_FEE: Record<string, DefinitionCall> = {
+  query_weight_to_fee: {
+    description: 'Query the output of the current WeightToFee given some input',
+    params: [
+      {
+        name: 'weight',
+        type: 'Weight'
+      }
+    ],
+    type: 'Balance'
+  }
+};
+
+const V3_QUERY_LENGTH_TO_FEE: Record<string, DefinitionCall> = {
+  query_length_to_fee: {
+    description: 'Query the output of the current LengthToFee given some input',
+    params: [
+      {
+        name: 'length',
+        type: 'u32'
+      }
+    ],
+    type: 'Balance'
+  }
+};
+
 export const runtime: DefinitionsCall = {
   TransactionPaymentApi: [
     {
-      methods: objectSpread({
-        query_info: {
-          description: 'The transaction info',
-          params: [
-            {
-              name: 'uxt',
-              type: 'Extrinsic'
-            },
-            {
-              name: 'len',
-              type: 'u32'
-            }
-          ],
-          type: 'RuntimeDispatchInfo'
-        }
-      }, V1_V2_SHARED_PAY),
+      methods: objectSpread(
+        {},
+        V3_QUERY_WEIGHT_TO_FEE,
+        V3_QUERY_LENGTH_TO_FEE,
```

Comment:
... and this one is only for `TransactionPaymentCallApi` (so the naming mismatches)